docs/architecture/adr-030-consensus-refactor.md

```diff
@@ -126,6 +126,184 @@ func TestConsensusXXX(t *testing.T) {
 }
 ```
 
+
+## Consensus Executor
+
+## Consensus Core
+
+```go
+type Event interface{}
+
+type EventNewHeight struct {
+    Height           int64
+    ValidatorSetSize int
+    ValidatorId      int
+}
+
+type EventNewRound struct {
+    Height           int64
+    Round            int
+}
+
+type EventProposal struct {
+    Height           int64
+    Round            int
+    Timestamp        Time
+    BlockID          BlockID
+    POLRound         int
+    Sender           int   
+}
+
+type Majority23PrevotesBlock struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+}
+
+type Majority23PrecommitBlock struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+}
+
+type Majority23PrevotesAny struct {
+    Height           int64
+    Round            int
+}
+
+type Majority23PrecommitAny struct {
+    Height           int64
+    Round            int
+}
+
+type TimeoutPropose struct {
+    Height           int64
+    Round            int
+}
+
+type TimeoutPrevotes struct {
+    Height           int64
+    Round            int
+}
+
+type TimeoutPrecommit struct {
+    Height           int64
+    Round            int
+}
+
+
+type Message interface{}
+
+type MessageProposal struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+    POLRound         int
+}
+
+type VoteType int
+
+const (
+	VoteTypeUnknown VoteType = iota
+	Prevote
+	Precommit
+)
+
+
+type MessageVote struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+    Type             VoteType
+}
+
+
+type MessageDecision struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+}
+
+type TriggerTimeout struct {
+    Height           int64
+    Round            int
+    Duration         Duration
+}
+
+
+type RoundStep int
+
+const (
+	RoundStepUnknown RoundStep = iota
+	RoundStepPropose       
+	RoundStepPrevote
+	RoundStepPrecommit
+	RoundStepCommit
+)
+
+type State struct {
+	Height           int64
+	Round            int
+	Step             RoundStep
+	LockedValue      BlockID
+	LockedRound      int
+	ValidValue       BlockID
+	ValidRound       int
+	ValidatorSetSize int
+	ValidatorId      int
+}
+
+func proposer(height int64, round int) int {}
```

Comment:
I'm not sure how to structure this function given that the validator set is changing over time ...